### PR TITLE
Funnel all etcd requests through EtcdClient::GenericRequest.

### PR DESCRIPTION
**Pull request by pphaneuf**

### CODE REVIEW

File: cpp/util/etcd.cc [hunk 1]
```diff
@@ -677,9 +660,15 @@ Status EtcdClient::WatchState::HandleSingleValueRequestDone(
...
-                                         const shared_ptr<JsonObject>& json,
-                                         int64_t etcd_index) {
+void EtcdClient::WatchState::RequestDone(GenericResponse* gen_resp,
+                                         Task* task) {
+  // We clean up this way instead of using util::Task::DeleteWhenDone,
+  // because our task is long-lived, and we do not want to accumulate
+  // these objects.
+  unique_ptr<GenericResponse> gen_resp_deleter(gen_resp);
+
+  LOG(INFO) << "etcd_index: " << gen_resp->etcd_index;
```
Comment (AlCutter) [inline cpp/util/etcd.cc]: `VLOG` I suspect :)

File: cpp/util/etcd.cc [hunk 2]
```diff
@@ -1038,19 +1023,9 @@ void EtcdClient::Watch(const string& key, const WatchCallback& cb,
...
-  resp->etcd_index = etcd_index;
-  resp->json_body = json_body;
-  task->Return(status);
-}
-
-
 void EtcdClient::Generic(const string& key, const map<string, string>& params,
-                         evhttp_cmd_type verb, GenericResponse* resp,
-                         Task* task) {
+                         evhttp_cmd_type verb, bool separate_conn,
```
Comment (AlCutter) [inline cpp/util/etcd.cc]: `separate_conn` might be nicer as an `enum` for readability at call sites, unless, you know, this just happens to be going away RSN :)

File: cpp/util/etcd_test.cc [hunk 3]
```diff
@@ -21,8 +21,8 @@ using std::pair;
 using std::placeholders::_1;
 using std::placeholders::_2;
 using std::placeholders::_3;
-using std::placeholders::_4;
 using std::placeholders::_5;
+using std::placeholders::_6;
```
Comment (AlCutter) [inline cpp/util/etcd_test.cc]: Are we going to run out of these ? :)

**Comment by AlCutter** [comment]
LGTM with nits

File: cpp/util/etcd_test.cc (hunk 3 above)
Comment (pphaneuf) [inline cpp/util/etcd_test.cc]: In my libstdc++ headers, there's 29. :smiley:

File: cpp/util/etcd.cc (hunk 2 above)
Comment (pphaneuf) [inline cpp/util/etcd.cc]: Going away RSN, I'll stick with the ```bool``` for now... :wink:

File: cpp/util/etcd.cc (hunk 1 above)
Comment (pphaneuf) [inline cpp/util/etcd.cc]: A ```VLOG(2)``` even. I kind of forgot that in, could have deleted it, but what the heck.

**Pull request merged**